lib/cretonne/src/ir/instructions.rs

```diff
@@ -333,6 +333,56 @@ impl InstructionData {
         }
     }
 
+    /// Get a mutable reference to the value list of jump arguments of this branch.
+    ///
+    /// Multi-destination branches like `br_table` return `None`.
+    /// TODO: deal with multi-branch instructions
+    pub fn branch_arguments_mut(&mut self, dest: Ebb) -> Option<&mut ValueList> {
```

Comment:
This is a little dangerous because the returned value list contains *all* of the instruction's arguments, including one or two values controlling the branch. So it is not just the EBB arguments.

There is already a `take_value_list()` method which can be used for manipulating these lists. See `legalizer/split.rs` for an example. We could generate a `value_list_mut()` method, if that matches your needs better.